tests/unit/src/test_formationPattern.cpp

```diff
@@ -0,0 +1,161 @@
+/* test_formationPattern.cpp
+Copyright (c) 2021 by Peter van der Meer
+
+Endless Sky is free software: you can redistribute it and/or modify it under the
+terms of the GNU General Public License as published by the Free Software
+Foundation, either version 3 of the License, or (at your option) any later version.
+
+Endless Sky is distributed in the hope that it will be useful, but WITHOUT ANY
+WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A
+PARTICULAR PURPOSE. See the GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License along with
+this program. If not, see <https://www.gnu.org/licenses/>.
+*/
+
+#include "es-test.hpp"
+
+// Include only the tested class's header.
+#include "../../../source/FormationPattern.h"
+
+// Include a helper for creating well-formed DataNodes (to use for loading formations).
+#include "datanode-factory.h"
+
+// ... and any system includes needed for the test file.
+#include <map>
+#include <string>
+
+
```

Comment:
Not sure if this is the standard amount of newlines here (and on phone so cant really check)

Reply:
The other tests only seem to have 1 newline here. I'll remove one.